core/domain/user_services.py

```diff
@@ -694,25 +728,26 @@ def has_fully_registered(user_id):
         feconf.REGISTRATION_PAGE_LAST_UPDATED_UTC)
 
 
-def create_new_user(user_id, email):
+def create_new_user(gae_user_id, email):
     """Creates a new user.
 
     Args:
-        user_id: str. The unique ID of the user.
+        gae_user_id: str. The unique GAE user ID of the user.
         email: str. The user email.
 
     Returns:
         UserSettings. The newly-created user settings domain object.
 
     Raises:
-        Exception: If a user with the given user_id already exists.
+        Exception: If a user with the given gae_user_id already exists.
     """
-    user_settings = get_user_settings(user_id, strict=False)
+    user_settings = get_user_settings(gae_user_id, strict=False)
     if user_settings is not None:
-        raise Exception('User %s already exists.' % user_id)
+        raise Exception('User %s already exists.' % gae_user_id)
 
+    user_id = gae_user_id
```

Comment:
Shouldn't the user_id be different from the gae_user_id? At least it needs some kind of TODO/issue, right?

Reply:
Forgot to add a comment. Done.